lib/puppetserver/ca/generate_action.rb

```diff
@@ -0,0 +1,250 @@
+require 'optparse'
+require 'openssl'
+require 'puppetserver/utils/file_utils'
+require 'puppetserver/settings/ttl_setting'
+
+module Puppetserver
+  module Ca
+    class GenerateAction
+      CA_EXTENSIONS = [
+        ["basicConstraints", "CA:TRUE", true],
+        ["keyUsage", "keyCertSign, cRLSign", true],
+        ["subjectKeyIdentifier", "hash", false],
+        ["authorityKeyIdentifier", "keyid:always", false]
+      ]
```

Comment:
I copied these from @justinstoller https://github.com/puppetlabs/puppetserver/blob/a345d592eeee881bf8d1f4265239eeb65d5fae18/acceptance/lib/puppet_x/acceptance/pki.rb#L9  But I don't really understand then enough to say how correct/tested they are.  I'm assuming since Justin wrote them they are PERFECT :)

Reply:
https://www.ibm.com/support/knowledgecenter/en/SSB23S_1.1.0.13/gtps7/cfgcert.html
and
https://security.stackexchange.com/questions/27797/what-damage-could-be-done-if-a-malicious-certificate-had-an-identical-subject-k
for reference